### PR TITLE
New StatusChip-component with reusable base statuses

### PR DESCRIPTION
**Pull request by denektenina**
# Description

This is a solution to two problems that became evident when I needed to use the existing StatusChip with a new icon. 

1) The existing StatusChip accepts a string representing the icon as a prop, meaning that if I need to use a new icon I have to modify StatusChip to accept a prop for the new icon. This violates the Open / Closed Principe and can cause an unnecessary messy component over time, since every icon that might be used only one places has to be a prop and a new if/else in the component.

2) The component takes props for icon, text, color and padding, meaning that even though we use the same component, they might look very different across the app, losing some of the point of using the same component.

-> To solve the first problem I propose modifying the StatusChip component to accept icon as `children` instead so that new icons can be used without having to modify the component each time. (New name: BaseStatusChip)

-> To solve the second problem I propose adding a new component (new StatusChip) that hard codes a number of base styles, so that the choice of color and icon for similar states is consolidated across the app. Because the use case can be very different across the app, I am leaving the text as a prop still, even though the styling is the same. I.e. in some cases something might be "ongoing" but in other cases it might be "pending" - but they both use a light blue chip with an hourglass.

StatusChip will use BaseStatusChip to ensure the same styling.

BaseStatusChip now adheres to the Open / Closed Principe. StatusChip does not, but BaseStatusChip can be used in the edge cases not covered in StatusChip, ensuring that it will rarely or never need to be modified.

# How to test

Not used in app yet, but can be tested by inserting into any page and verifying that the result looks as expected.

# Checklist

Ensure that the changes are aligned with the expectations of PO/designer before marking the PR as ready for review.

Also ensure that the following criterias are met:

- [x] The changes are working as expected
- [x] The changes are tested OK for different screen sizes
- [x] The changes are tested OK for a11y
- [x] Interactive elements have data-testids
- [x] I have done a QA of my own changes

_Note: some of these criterias may not always be relevant, and can simply be marked as completed._

## Summary by CodeRabbit

* **New Features**
  * Added a status chip component with visual indicators for different statuses: Waiting, InProgress, Completed, Success, and Closed
  * Each status displays a unique icon and colour for improved visual distinction

[Review Change Stack](https://app.coderabbit.ai/change-stack/BIBSYSDEV/NVA-Frontend/pull/8492)

### CODE REVIEW

File: src/components/_molecules/status-chip/StatusChip.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
+import BlockIcon from '@mui/icons-material/Block';
+import CheckIcon from '@mui/icons-material/Check';
+import HourglassEmptyIcon from '@mui/icons-material/HourglassEmpty';
+import LockClockIcon from '@mui/icons-material/LockClock';
+import { BaseStatusChip } from './BaseStatusChip';
+
+export enum StatusValue {
+  Waiting = 'WAITING',
```
Comment (iBuzza) [inline src/components/_molecules/status-chip/StatusChip.tsx]: ahould probably be pending, since that is what is most commonly used 
```suggestion
  Pending = 'PENDING',
```

**Review by iBuzza — DISMISSED** [review]
